examples/chat/client/cpp/chat.cc

```diff
@@ -158,26 +156,25 @@ int main(int argc, char** argv) {
   std::string ca_cert = oak::ApplicationClient::LoadRootCert(absl::GetFlag(FLAGS_ca_cert));
   LOG(INFO) << "Connecting to Oak Application: " << address;
 
-  RoomToken room_access_token;
-  if (!absl::Base64Unescape(absl::GetFlag(FLAGS_room_access_token), &room_access_token)) {
-    LOG(FATAL) << "Failed to parse --room_access_token as base64";
+  std::string room_secret = absl::GetFlag(FLAGS_room_secret);
+  std::string private_key;
+
+  // If no room secret, for retreiving a private key, was provided, create a fresh private/public
+  // key pair, and store the private key in a file for later use. A secure key sharing mechanims is
+  // needed to share this file with other clients.
+  if (room_secret.empty()) {
+    private_key = oak::ApplicationClient::GenerateKeyPair();
+    oak::ApplicationClient::StorePrivateKey(private_key, "chat-room.key");
+  } else {
+    private_key = oak::ApplicationClient::LoadPrivateKey(room_secret);
```

Comment:
Personally I would have avoided this extra logic and just required that the key is always pre-generated via oak_sign or some other mechanism, and this binary just consumes it. Otherwise now we need to keep track of the key it creates, and whether that overwrites other existing keys, and where to move the key, how to document it (the flag description does not mention the file name so it seems hard to figure out what is going to happen).

To clarify what I mean, I also see this a bit similar to how git uses a SSH key for auth, but it does not need to know how to generate and store those keys.

Reply:
That's fine, but that's not what I was talking about. I was just saying that this app should not have to generate the key itself, it can just assume it is already generated and available as a file somewhere.

Reply:
I understand, but in the meantime, we still need to generate the key-pair for the example to work. I have reworded the TODO comment.